x/staking/keeper/grpc_query.go

```diff
@@ -397,6 +397,10 @@ func (k Querier) HistoricalInfo(ctx context.Context, req *types.QueryHistoricalI
 		return nil, status.Error(codes.InvalidArgument, "height cannot be negative")
 	}
 
+	if has, err := k.storeService.OpenKVStore(ctx).Has(types.NextMigrateHistoricalInfoKey); has || err != nil {
+		return nil, status.Error(codes.Internal, "store migration has not finished, try again later")
+	}
```

Comment:
will this break any important services temporarily? do we know roughly how many keys that need to be migrated, and therefore roughly how long these services will be impacted?

Reply:
The max entires param is 10k. So 10k keys at 1k per block = 10 blocks. 6 secs per block = 1 min